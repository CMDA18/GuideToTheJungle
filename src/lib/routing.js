// @flow
import React from 'react'
import { Route, Redirect } from 'react-router'

type RedirectType = {
  to: string,
  from?: string,
  status?: number
}

export const RedirectWithStatus = ({ from, to, status }: RedirectType) => (
  <Route render={({ staticContext }: any) => { // eslint-disable-line flowtype/no-weak-types
    // there is no `staticContext` on the client, so
    // we need to guard against that here
    /* istanbul ignore next */
    if (staticContext) {
      staticContext.status = status
    }
    /* istanbul ignore next */
    return <Redirect from={from} to={to}/>
  }}/>
)
