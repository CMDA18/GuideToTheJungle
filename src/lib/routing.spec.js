import React from 'react'
import { RedirectWithStatus } from './routing'

it('renders without crashing', () => {
  expect(<RedirectWithStatus status={302} to={'/'} />).toMatchSnapshot()

  expect(RedirectWithStatus({
    from: '/',
    to: '/somewhere',
    status: 302
  })).toMatchSnapshot()
})
