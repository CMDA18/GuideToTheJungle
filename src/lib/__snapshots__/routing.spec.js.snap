// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<RedirectWithStatus
  status={302}
  to="/"
/>
`;

exports[`renders without crashing 2`] = `
<Route
  render={[Function]}
/>
`;
